### PR TITLE
Fix firmware download

### PR DESCRIPTION
**Pull request by Ninjeneer**
Two modifications :

1) Wrapped bucketStream `end` event in a Promise in order to avoid Express to send a 404 status (as the endpoint is full of callback, it seems that the default Express behavior is to send a 404 status)

2) Refactored endpoint route, because `/api/chargingstations/firmware/{filename}` could break with names containing a `/`.

### CODE REVIEW

**Review by jerome-benoit — CHANGES_REQUESTED** [review]
Have you tested that the old endpoint is still working?

File: src/types/requests/HttpChargingStationRequest.ts [hunk 1]
```diff
@@ -102,5 +102,5 @@ export interface HttpIsAuthorizedRequest {
 }
 
 export interface HttpChargingStationGetFirmwareRequest {
-  ID: string;
+  FileName: string;
```
[the remark this answers is not in the document]
Comment (Ninjeneer) [reply src/types/requests/HttpChargingStationRequest.ts]: Ok switched to lowerCamelCase.
However, some query params in the code are still in UpperCamelCase as it was before. Is it a migration in progress ?

[the remark this answers is not in the document]
Comment (LucasBrazi06) [reply src/types/requests/HttpChargingStationRequest.ts]: > Ok switched to lowerCamelCase.
> However, some query params in the code are still in UpperCamelCase as it was before. Is it a migration in progress ?

If it's a Querystring param it's Pascal case.

[the remark this answers is not in the document]
Comment (Ninjeneer) [reply src/types/requests/HttpChargingStationRequest.ts:105]: Ok, switched back to Pascal case then.

**Review by LucasBrazi06 — APPROVED** [review]

**Pull request merged**